commands/stats.ts

```diff
@@ -32,17 +38,26 @@ export const action = (message: Message, key: string) => {
       roleData.deleted ||
       roleName === "@everyone" ||
       roleName === "YAGPDB.xyz" ||
+      roleName === "Bot" ||
       roleCount === 0 ||
       roleName.toLowerCase().match(/([a-z][a-z][a-z][0-9][0-9][0-9][0-9])/g)
     )
       continue;
-
-    rollCounts.push(`\`${roleName} members: ${roleCount}\``);
+    
+    if (YEARS.indexOf(roleName) !== -1) rollCounts.year.push(`\`${roleName}: ${roleCount}\``);
+    else if (ASSOCIATIONS.indexOf(roleName) !== -1) rollCounts.assoc.push(`\`${roleName}: ${roleCount}\``);
+    else if (PROGRAMS.indexOf(roleName) !== -1) rollCounts.program.push(`\`${roleName}: ${roleCount}\``);
+    else if (PRONOUNS.indexOf(roleName) !== -1) rollCounts.pronoun.push(`\`${roleName}: ${roleCount}\``);
+    else rollCounts.misc.push(`\`${roleName}: ${roleCount}\``);
```

Comment:
We're using `Array.indexOf()` to organize the role counts as opposed to relying on Regex, because of this: https://stackoverflow.com/questions/5296268/fastest-way-to-check-a-string-contain-another-substring-in-javascript

(TLDR): 
<img src="https://user-images.githubusercontent.com/43019056/104871980-90e6d200-591a-11eb-949c-643dc26916a0.png" width="600"/>